barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.hpp

```diff
@@ -60,110 +118,152 @@ class GoblinTranslatorCircuitBuilder : CircuitBuilderBase<arithmetization::Gobli
-        ACCUMULATOR_HI_LIMBS_RANGE_CONSTRAINT_2,
-        ACCUMULATOR_HI_LIMBS_RANGE_CONSTRAINT_3,
-        ACCUMULATOR_HI_LIMBS_RANGE_CONSTRAINT_4,
-        ACCUMULATOR_HI_LIMBS_RANGE_CONSTRAINT_TAIL,
-        QUOTIENT_LO_BINARY_LIMBS, // Quotient limbs
-        QUOTIENT_HI_BINARY_LIMBS,
-        QUOTIENT_LO_LIMBS_RANGE_CONSTRAIN_0, // Range constraints for quotient
-        QUOTIENT_LO_LIMBS_RANGE_CONSTRAIN_1,
-        QUOTIENT_LO_LIMBS_RANGE_CONSTRAIN_2,
-        QUOTIENT_LO_LIMBS_RANGE_CONSTRAIN_3,
-        QUOTIENT_LO_LIMBS_RANGE_CONSTRAIN_4,
-        QUOTIENT_LO_LIMBS_RANGE_CONSTRAIN_TAIL,
-        QUOTIENT_HI_LIMBS_RANGE_CONSTRAIN_0,
-        QUOTIENT_HI_LIMBS_RANGE_CONSTRAIN_1,
-        QUOTIENT_HI_LIMBS_RANGE_CONSTRAIN_2,
-        QUOTIENT_HI_LIMBS_RANGE_CONSTRAIN_3,
-        QUOTIENT_HI_LIMBS_RANGE_CONSTRAIN_4,
-        QUOTIENT_HI_LIMBS_RANGE_CONSTRAIN_TAIL,
+        ACCUMULATORS_BINARY_LIMBS_3,              // Highest limb is 50 bits (254 mod 68)
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_0, // Range constraints for the current accumulator limbs (no need to
+                                                  // redo previous accumulator)
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_1,
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_2,
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_3,
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_4,
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_TAIL,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_0,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_1,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_2,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_3,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_4,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_TAIL,
+        QUOTIENT_LOW_BINARY_LIMBS, // Quotient limbs
+        QUOTIENT_HIGH_BINARY_LIMBS,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAIN_0, // Range constraints for quotient
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAIN_1,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAIN_2,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAIN_3,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAIN_4,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAIN_TAIL,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAIN_0,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAIN_1,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAIN_2,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAIN_3,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAIN_4,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAIN_TAIL,
         RELATION_WIDE_LIMBS, // Limbs for checking the correctness of  mod 2²⁷² relations. TODO(kesha): add range
                              // constraints
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_0,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_1,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_2,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_3,
+
+        TOTAL_COUNT
 
     };
-    static constexpr size_t MAX_OPERAND = 3;
+
+    // Basic goblin translator has the minicircuit size of 2048, so optimize for that case
```

Comment:
Can you say more about what this means?

Reply:
In the proving system we have the full circuit which is 16 times larger, but it is only used for the permutation argument. The minicircuit has all the other logic